packages/flutter_tools/test/general.shard/build_system/targets/icon_tree_shaker_test.dart

```diff
@@ -0,0 +1,517 @@
+  BufferLogger logger;
+  MemoryFileSystem fs;
+  MockProcessManager mockProcessManager;
+  MockProcess fontSubsetProcess;
+  MockArtifacts mockArtifacts;
+  DevFSStringContent fontManifestContent;
+
+  const String dartPath = '/flutter/dart';
+  const String constFinderPath = '/flutter/const_finder.snapshot.dart';
+  const String fontSubsetPath = '/flutter/font-subset';
+
+  const String inputPath = '/input/fonts/MaterialIcons-Regular.ttf';
+  const String outputPath = '/output/fonts/MaterialIcons-Regular.ttf';
+  const String relativePath = 'fonts/MaterialIcons-Regular.ttf';
+
+  List<String> getConstFinderArgs(String appDillPath) => <String>[
+    dartPath,
+    constFinderPath,
+    '--kernel-file', appDillPath,
+    '--class-library-uri', 'package:flutter/src/widgets/icon_data.dart',
+    '--class-name', 'IconData',
+  ];
+
+  const List<String> fontSubsetArgs = <String>[
+    fontSubsetPath,
+    outputPath,
+    inputPath,
+  ];
+
+  void _addConstFinderInvocation(
+    String appDillPath, {
+    int exitCode = 0,
+    String stdout = '',
+    String stderr = '',
+  }) {
+    when(mockProcessManager.run(getConstFinderArgs(appDillPath))).thenAnswer((_) async {
+      return ProcessResult(0, exitCode, stdout, stderr);
+    });
+  }
+
+  void _resetFontSubsetInvocation({
+    int exitCode = 0,
+    String stdout = '',
+    String stderr = '',
+    @required mocks.CompleterIOSink stdinSink,
+  }) {
+    assert(stdinSink != null);
+    stdinSink.writes.clear();
+    when(fontSubsetProcess.exitCode).thenAnswer((_) async => exitCode);
+    when(fontSubsetProcess.stdout).thenAnswer((_) => Stream<List<int>>.fromIterable(<List<int>>[utf8.encode(stdout)]));
+    when(fontSubsetProcess.stderr).thenAnswer((_) => Stream<List<int>>.fromIterable(<List<int>>[utf8.encode(stderr)]));
+    when(fontSubsetProcess.stdin).thenReturn(stdinSink);
+    when(mockProcessManager.start(fontSubsetArgs)).thenAnswer((_) async {
+      return fontSubsetProcess;
+    });
+  }
+
+  setUp(() {
+    fontManifestContent = DevFSStringContent(validFontManifestJson);
+
+    mockProcessManager = MockProcessManager();
+    fontSubsetProcess = MockProcess();
+    mockArtifacts = MockArtifacts();
+
+    fs = MemoryFileSystem();
+    logger = BufferLogger(
+      terminal: AnsiTerminal(
+        stdio: mocks.MockStdio(),
+        platform: _kNoAnsiPlatform,
+      ),
+      outputPreferences: OutputPreferences.test(showColor: false),
+    );
+
+    fs.file(constFinderPath).createSync(recursive: true);
+    fs.file(dartPath).createSync(recursive: true);
+    fs.file(fontSubsetPath).createSync(recursive: true);
+    when(mockArtifacts.getArtifactPath(Artifact.constFinder)).thenReturn(constFinderPath);
+    when(mockArtifacts.getArtifactPath(Artifact.fontSubset)).thenReturn(fontSubsetPath);
+    when(mockArtifacts.getArtifactPath(Artifact.engineDartBinary)).thenReturn(dartPath);
+  });
+
+  Environment _createEnvironment(Map<String, String> defines) {
+    return Environment(
```

Comment:
Use Environment.test

Reply:
Environment.test wants a test directory, which I don't have, right?

Reply:
It allows you to set a single directory that everything is based on, if the directories themselves don't matter much

Reply:
Ah, I get it now, done.